src/utils/data_types.py

```diff
@@ -0,0 +1,272 @@
+
+class SingleSelectType(Enum):
+    STATUS = "Status"
+    PRIORITY = "Priority"
+    SIZE = "Size"
+    ITERATION = "Iteration"
+    SECTION = "Section"
+
+
+@dataclass
+class ProjectItemEvent:
+    node_id: str
+    sender: str
+
+    """
+    Interface method to process the event and optionally return message to be posted in Discord.
+    """
+
+    async def process(
+        self,
+        user_text_mention: str,
+        post: GuildPublicThread,
+        client: RESTClientImpl,
+        logger: Logger,
+        shared_forum_channel: SharedForumChannel,
+        forum_channel_id: int,
+    ) -> str | GuildForumChannel | None:
+        pass
+
+
+class SimpleProjectItemEvent(ProjectItemEvent):
+    def __init__(self, node_id: str, sender: str, action_type: str):
+        super().__init__(node_id, sender)
+        self.event_type = SimpleProjectItemEventType(action_type)
+
+    async def process(
+        self,
+        user_text_mention: str,
+        post: GuildPublicThread,
+        client: RESTClientImpl,
+        logger: Logger,
+        _shared_forum_channel: SharedForumChannel,
+        _forum_channel_id: int,
+    ) -> str | None:
+        match self.event_type.value:
+            case "archived":
+                message = f"Task zarchiwizowany przez: {user_text_mention}."
+                await client.edit_channel(post.id, archived=True)
+                logger.info(f"Post {self.node_id} archived.")
+                return message
+            case "restored":
+                message = f"Task przywrócony przez: {user_text_mention}."
+                await client.edit_channel(post.id, archived=False)
+                logger.info(f"Post {self.node_id} restored.")
+                return message
+            case "deleted":
+                await client.delete_channel(post.id)
+                logger.info(f"Post {self.node_id} deleted.")
+                return None
+            case _:
+                return None
+
+
+class ProjectItemEditedBody(ProjectItemEvent):
+    def __init__(self, node_id: str, editor: str, new_body: str):
+        super().__init__(node_id, editor)
+        self.new_body = new_body
+
+    async def process(
+        self,
+        user_text_mention: str,
+        _post: GuildPublicThread,
+        client: RESTClientImpl,
+        logger: Logger,
+        _shared_forum_channel: SharedForumChannel,
+        forum_channel_id: int,
+    ) -> str:
+        message = f"Opis taska zaktualizowany przez: {user_text_mention}. Nowy opis: \n{self.new_body}"
+        logger.info(f"Post {self.node_id} body updated.")
+
+        return message
+
+
+class ProjectItemEditedAssignees(ProjectItemEvent):
+    def __init__(self, node_id: str, editor: str, new_assignees: list[str]):
+        super().__init__(node_id, editor)
+        self.new_assignees = new_assignees
+
+    async def process(
+        self,
+        user_text_mention: str,
+        post: GuildPublicThread,
+        client: RESTClientImpl,
+        logger: Logger,
```

Comment:
I am unsure if passing logger everywhere as an argument is considered good practice...